Docs: Notes on adding an icon library (#278)
ideditor/schema-builder: docs/Adding-a-new-icon-source.md

# Adding a new icon source (maintainers)

> This document describes what has to happen when the tagging schema adopts icons from a **new family** (a new string prefix). The latest addition in practice was **`roentgen-`** ([Röntgen](https://github.com/enzet/map-machine#r%C3%B6ntgen-icon-set) icons; see also [`ICONS.md`](../ICONS.md)). It complements the contributor-facing overview there.

Reviewer: ```suggestion
This document describes what has to happen when the tagging schema adopts icons from a **new family** (a new string prefix). See [`ICONS.md`](../ICONS.md) for already added ones. It complements the contributor-facing overview there.
```

we really should not add parts doomed to get stale

> ## Is this a breaking change to the JSON schema?
> 
> **No**, we do not consider a new icon prefix a breaking schema change.
> 
> In [`schemas/preset.json`](../schemas/preset.json), `icon` is a free-form string, so values such as `roentgen-ford` need no schema update.
> 
> Apps and libraries should fall back when an icon is unknown or missing (e.g. generic glyph or text only); with that in place, new prefixes should not break integrations; the experience is simply weaker until explicit support exists.

Reviewer: ```suggestion
## Is this a breaking change to the JSON schema?

**Yes**, as some of already existing presets may start using icons from the new set.

In [`schemas/preset.json`](../schemas/preset.json), `icon` is a free-form string, so values such as `roentgen-ford` need no schema update.

Apps and libraries should fall back when an icon is unknown or missing (e.g. generic glyph or text only); with that in place, new prefixes should not break integrations entirely; the experience is simply weaker until explicit support exists.
```

I disagree whether it counts or should count as a breaking change

## End-to-end flow

1. **Choose a prefix** (e.g. `roentgen-`) and ensure **stable public SVG URLs** (GitHub raw, npm package, etc.).

> 2. **id-tagging-schema**: use `"icon": "prefix-name"` in presets, categories, or field `icons`.

Reviewer: `2. **id-tagging-schema**: use `"icon": "prefix-name"` in presets, categories, or field `icons`.`

that surely should be the last step? or at least after 

> 4. **id-tagging-schema**: bump the schema-builder devDependency and rebuild.

3. **schema-builder**: document the source; add Taginfo `icon_url` mapping in `lib/build.js`; release `@ideditor/schema-builder`.
4. **id-tagging-schema**: bump the schema-builder devDependency and rebuild.

> 5. **Each editor / consumer** that renders preset icons: extend its own resolution or bundling (see below).

Reviewer: maybe we should contact data consumers and let them know about it? Or is mentioning in release notes sufficient?

| [`lib/build.js`](../lib/build.js) | In `generateTaginfo`, add an `else if` for the new prefix so Taginfo preset items get a correct `icon_url` (same idea as `maki-`, `temaki-`, `fas-`, `iD-`). **Note:** `roentgen-` is used in presets but was historically missing from some paths—new families should not repeat that gap. |
| Tests | Optional: extend `tests/schema-builder.test.js` with a fixture icon to assert Taginfo output. |

## 3. iD

iD shows preset icons from **SVG symbol sprites** built at release time and shipped with the app.

| Area | Action |
|------|--------|
| `package.json` | Add npm dependency if the set is published, **or** vendor SVGs under e.g. `svg/…`. |
| `dist:svg:…` | New `svg-sprite` script, same pattern as `dist:svg:maki` / `dist:svg:temaki`. |
| `modules/svg/defs.js` | Register the new sprite sheet id. |
| `modules/ui/preset_icon.js` | Only if the set needs special sizing/styling (e.g. Röntgen). |
| `dist/img/*-sprite.svg` | Generated by the sprite build. |

[Rapid](https://github.com/facebook/Rapid) follows the same sprite approach where it mirrors iD.

## 4. StreetComplete

> 

Reviewer: we should here link SC docs if they exist, not make backup of them
